ledger/accountdb_test.go

```diff
@@ -3761,3 +3761,104 @@ func TestUnfinishedCatchpointsTable(t *testing.T) {
 	}
 	require.Equal(t, expected, ret)
 }
+
+func TestRemoveOfflineStateProofID(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	dbs, _ := dbOpenTest(t, true)
+	setDbLogging(t, dbs)
+	defer dbs.Close()
+
+	tx, err := dbs.Wdb.Handle.Begin()
+	require.NoError(t, err)
+	defer tx.Rollback()
+
+	accts := ledgertesting.RandomAccounts(20, true)
+	for addr, acct := range accts {
+		rand.Read(acct.StateProofID[:])
+		accts[addr] = acct
+	}
+
+	// this is the same seq as accountsInitTest makes but it stops
+	// before the online accounts table creation to generate a trie and commit it
+	_, err = accountsInit(tx, accts, config.Consensus[protocol.ConsensusCurrentVersion])
+	require.NoError(t, err)
+
+	err = accountsAddNormalizedBalance(tx, config.Consensus[protocol.ConsensusCurrentVersion])
+	require.NoError(t, err)
+
+	err = accountsCreateResourceTable(context.Background(), tx)
+	require.NoError(t, err)
+
+	err = performResourceTableMigration(context.Background(), tx, nil)
+	require.NoError(t, err)
+
+	// create account hashes
+	getRootHash := func(commit bool) (crypto.Digest, error) {
+		rows, err := tx.Query("SELECT address, data FROM accountbase")
+		require.NoError(t, err)
+		defer rows.Close()
+
+		mc, err := MakeMerkleCommitter(tx, false)
+		require.NoError(t, err)
+		trie, err := merkletrie.MakeTrie(mc, TrieMemoryConfig)
+		require.NoError(t, err)
+
+		var addr basics.Address
+		for rows.Next() {
+			var addrbuf []byte
+			var encodedAcctData []byte
+			err = rows.Scan(&addrbuf, &encodedAcctData)
+			require.NoError(t, err)
+			copy(addr[:], addrbuf)
+			var ba baseAccountData
+			err = protocol.Decode(encodedAcctData, &ba)
+			require.NoError(t, err)
+			addHash := accountHashBuilderV6(addr, &ba, encodedAcctData)
+			added, err := trie.Add(addHash)
+			require.NoError(t, err)
+			require.True(t, added)
+		}
+		if commit {
+			trie.Evict(true)
+		}
+		return trie.RootHash()
+	}
+	oldRoot, err := getRootHash(true)
```

Comment:
You never call getRootHash(false) but I am not really sure what Evict would do anyway

Reply:
true, I used to used it in both old/new but changed, i'll update

Reply:
fixed